### PR TITLE
hostboot: Add a printed out banner add startup

### PR DESCRIPTION
**Pull request by wak-google**
This better clarifies the ImageId immediately to the end user.

---
This change is [Reviewable](https://reviewable.io/reviews/open-power/hostboot/90)

### CODE REVIEW

**Comment by stewart-ibm** [comment]
@wak-google You may also be interested in https://patchwork.ozlabs.org/patch/730916/ which is where we're trying to sort out a way to expose all of this to the kernel (and userspace) so that programmatic decisions can be made (e.g. to upgrade firmware or not), or to, well, at least, reliably create a version number screen in petitboot.

**Comment by wak-google** [comment]
That would be great. For rhesus we were adding devicetree entries as part
of our hostboot which contained the version strings. We are now parsing
this out of our read-only powernvflash device directly for both rhesus and
zaius, by reading the VERSION partition and parsing the contents. It would
make sense to read this info from a standard node in the device tree
instead of parsing the bios image. As long as this is trivially extensible,
and at a glance it looks like it will be, we would be happy to take
advantage of it.

**Comment by stewart-ibm** [comment]
"William A. Kennington III" <notifications@github.com> writes:
> That would be great. For rhesus we were adding devicetree entries as part
> of our hostboot which contained the version strings. We are now parsing
> this out of our read-only powernvflash device directly for both rhesus and
> zaius, by reading the VERSION partition and parsing the contents. It would
> make sense to read this info from a standard node in the device tree
> instead of parsing the bios image. As long as this is trivially extensible,
> and at a glance it looks like it will be, we would be happy to take
> advantage of it.

Excellent!

I am thinking we're going to need an OPAL call or something to get
information on what firmware is on flash or on alternate sides (A/B,
Golden/primary) as these can change at runtime out-of-band. THe only
thing we can represent in the device tree is what we booted with.

FWIW current P8 Hostboot does read the VERSION partition and whack it in
the device tree, it's just that the contents of that node isn't
predictable enough to be useful for our purposes (such as comparing the
current version number to a "Latest firmware version" retreived over the
network)

-- 
Stewart Smith
OPAL Architect, IBM.

**Comment by dcrowell77** [comment]
Pulled in locally for test and review

File: src/usr/console/daemon.C [hunk 1]
```diff
@@ -70,6 +72,13 @@ namespace CONSOLE
             Uart::g_device->initialize();
         }
 
+        // Display a banner denoting the hostboot version
+        char banner[256];
+        snprintf(greeting, sizeof(greeting),
```
Comment (dcrowell77) [inline src/usr/console/daemon.C]: FYI - This doesn't compile, looks like you switched some var names at the last minute.

daemon.C: In function 'void* CONSOLE::consoleDaemon(void*)':
daemon.C:77:18: error: 'greeting' was not declared in this scope
         snprintf(greeting, sizeof(greeting),

I'll fix this up in my local push assuming no other issues.

**Pull request closed without merging**